server/routes/books-list.routes.js

```diff
@@ -3,34 +3,102 @@ const Book = require('../models/Book')
 const router = Router() 
 
 router.get(
-    '/show-books',
+    '/show-books/:id',
     async (req, res) => {
         try {
-            const {limit, page} = req.query
-            const books = await Book.find()
-            res.json({
-                data: books.slice(limit * (page - 1), limit * page), 
-                totalCount: books?.length - 1 || 0 
+            const { limit, page, sortField, sortOrder, filter, searchText } = req.query
+            const { id } = req?.params
+
+            let books = id.length > 1 ? await Book.find({ sellerId: id }) : await Book.find()
+            let filteredBooks = [];
+            
+            if (!!searchText) {
+                filteredBooks = books.filter(book => {
+                    return book.bookName.toLowerCase().includes(searchText.toLowerCase().trim()) 
+                    || book.author.toLowerCase().includes(searchText.toLowerCase().trim())
+                })
+                books = filteredBooks?.length ? filteredBooks : []
+            }
+            
+            if (!!filter) {
+                const parseFilter = filter.split(';').map(option => option.split('='))
+                parseFilter.map(option => {
```

Comment:
separate thi function to smaller function and moove it to `utils` file

Reply:
fixed